### DISCUSSION TITLE
CSV download to include all the data from all the reports

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#1385] [Anything Else]
We plan to start the work on this during this week. 

The idea is for the [current export CSV](https://plausible.io/docs/export-stats) functionality to include the ability to download all the other data from other reports too. At the moment, it only downloads the metrics from the top chart. 

Ideally the UI would stay the same but with that one click you would be able to for instance get a zip file that features individual CSV files for each individual report (referrers, pages, countries etc).

**Comment by Dev10000** [comment]
Yes, that's desperately needed as soon as possible so more advanced statistical calculations can be done in excel. One CSV file including all data columns would be the best.

For example, the Plausible dashboard average bounce rate or visit duration of the data set can be miss leading. Median, mode and distribution chart would give a better picture of the data in some cases.

Before you get this feature public, is there any way now to get the monthly data that has all the information including date-time, source, page URL, bounce rate, visit duration, top pages, entry pages, exit pages, country, device size, browser, os?

**Reply by metmarkosaric** [reply]
thanks @TimRNR! we're working on this now and hoping to release it to the public later this month if everything goes well

in the meanwhile, the best way to get all the stats is using [our stats API](https://plausible.io/docs/stats-api)

**Reply by ukutaht** [reply]
Just a note that the export we are working on will contain aggregated data only.

To calculate distribution charts for bounce rate, for example, you would need access to the raw clickstream data I believe. This will not be part of the export. The export will contain exactly what you see in the dashboard already, just in the CSV format.

**Reply by Dev10000** [reply]
Yes, good point, I need the raw clickstream data indeed to calculate any of the statistical calculations, median, mode and distribution. I think I can not get the raw data via API either. Is there any chance of getting the RAW data from somewhere?

We have an issue with the advertising platform as we are getting many fraudulent clicks and the plausible shows average bounce rate 70-80% and visit duration 30-50s but if we would calculate the median, mode or distribution it would probably show 100% and 1s, meaning a majority of the clicks are 100% bounce rate and 1s visit duration and the distribution would show exact distribution how bad the advertising traffic is.

For example, let's say we have 10 visitors on one page, which 9 visits are 1s and 1 visit is 20 min, the average would show 2 min time on the page that would not look too bad but the median, mode or distribution would tell a totally different story, showing 90% of clicks are 1s.

**Reply by metmarkosaric** [reply]
Self-hosting Plausible is the only way to get the raw data at this time. By self-hosting you also manage the Clickhouse database itself which is where you can get access to the raw data. We plan to allow for export/import between self-hosted and cloud which is when it might be possible to get raw data from the cloud too but this is still early stage and hasn't been determined yet.

About bounce rate and time on page - we plan to work on introducing the beacon API to measure those which would allow us to have a much more accurate method than the GA method. We'll be able to measure even visit duration for those with a single page session rather than setting them to 0 like the current standards our. Keep an eye on the thread here #190

**Comment by metmarkosaric** [comment]
You can now export all of your stats in a CSV format. Click on the download icon on the top right of your Plausible dashboard. It will export a ZIP file which includes CSV files of the individual reports. You can then import your stats into Excel or other data analysis tools. Learn more here: https://plausible.io/docs/export-stats

**Reply by MarcoGlauser** [reply]
Awesome, thank you! Do you have a rough timeline when this will be available for a self hosted release?
